common/explorer_api/src/gather.rs

```diff
@@ -0,0 +1,174 @@
+        JsonType::Integer,
+    ),
+    (
+        "libp2p_peer_connected_peers_total",
+        "network_peers_connected",
+        JsonType::Integer,
+    ),
+    (
+        "libp2p_outbound_bytes",
+        "network_libp2p_bytes_total_transmit",
+        JsonType::Integer,
+    ),
+    (
+        "libp2p_inbound_bytes",
+        "network_libp2p_bytes_total_receive",
+        JsonType::Integer,
+    ),
+    (
+        "notifier_head_slot",
+        "sync_beacon_head_slot",
+        JsonType::Integer,
+    ),
+    ("sync_eth2_synced", "sync_eth2_synced", JsonType::Boolean),
+];
+
+pub const VALIDATOR_PROCESS_METRICS: &[(&str, &str, JsonType)] = &[
+    (
+        "vc_validators_enabled_count",
+        "validator_active",
+        JsonType::Integer,
+    ),
+    (
+        "vc_validators_total_count",
+        "validator_total",
+        JsonType::Integer,
+    ),
+    (
+        "sync_eth2_fallback_configured",
+        "sync_eth2_fallback_configured",
+        JsonType::Boolean,
+    ),
+    (
+        "sync_eth2_fallback_connected",
+        "sync_eth2_fallback_connected",
+        JsonType::Boolean,
+    ),
+];
+
+/// Represents the type for the JSON output.
+#[derive(Debug, Clone)]
+pub enum JsonType {
+    Integer,
+    Boolean,
+}
+
+lazy_static! {
+    /// HashMap representing the `BEACON_PROCESS_METRICS`.
+    pub static ref BEACON_METRICS_MAP: HashMap<String, (String, JsonType)> = BEACON_PROCESS_METRICS
+        .iter()
+        .map(|(k, v, ty)| (k.to_string(), (v.to_string(), ty.clone())))
+        .collect();
+    /// HashMap representing the `VALIDATOR_PROCESS_METRICS`.
+    pub static ref VALIDATOR_METRICS_MAP: HashMap<String, (String, JsonType)> =
+        VALIDATOR_PROCESS_METRICS
+            .iter()
+            .map(|(k, v, ty)| (k.to_string(), (v.to_string(), ty.clone())))
+            .collect();
+}
+
+/// Returns the value from a Counter/Gauge `MetricType` assuming that it has no associated labels
+/// else it returns `None`.
+fn get_value(mf: &MetricFamily) -> Option<i64> {
+    let metric = mf.get_metric().first()?;
+    match mf.get_field_type() {
+        MetricType::COUNTER => Some(metric.get_counter().get_value() as i64),
+        MetricType::GAUGE => Some(metric.get_gauge().get_value() as i64),
+        _ => None,
+    }
+}
+
+/// Return a json value given the JsonType.
+fn get_typed_value(value: i64, ty: &JsonType) -> serde_json::Value {
+    match ty {
+        JsonType::Integer => json!(value),
+        JsonType::Boolean => {
+            if value > 0 {
+                json!(true)
+            } else {
+                json!(false)
+            }
+        }
+    }
+}
+/// Collects all metrics and returns a `serde_json::Value` object with the required metrics
+/// from the metrics hashmap.
+pub fn gather_metrics(
+    metrics_map: &HashMap<String, (String, JsonType)>,
+) -> Result<serde_json::Value, String> {
+    let metric_families = lighthouse_metrics::gather();
+    let mut res = serde_json::Map::new();
```

Comment:
Perhaps we could avoid some resizing allocations with this:

```suggestion
    let mut res = serde_json::Map::with_capacity(metrics_map.len());
```